packages/sources/wrapped/src/endpoint/deposits.ts

```diff
@@ -0,0 +1,52 @@
+import { Requester, Validator, AdapterError } from '@chainlink/ea-bootstrap'
+import { Config, ExecuteWithConfig, InputParameters, AxiosResponse } from '@chainlink/types'
+
+export const supportedEndpoints = ['deposits']
+
+export interface ResponseSchema {
+  [token: string]: Address[]
+}
+
+export type Address = {
+  address: string
+}
+
+const customError = (data: any) => {
+  return typeof data !== 'object'
+}
+
+export const inputParameters: InputParameters = {
+  symbol: false,
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
+  const validator = new Validator(request, inputParameters)
+  if (validator.error) throw validator.error
+
+  const jobRunID = validator.validated.id
+  const symbol = validator.validated.data.symbol || 'ETH'
+  const url = `/deposits`
+
+  const options = { ...config.api, url }
+  const response = await Requester.request<ResponseSchema>(options, customError)
+  const addresses = response.data[symbol]
+
+  if (!addresses) {
+    const keys = Object.keys(response.data).join()
+    throw new AdapterError({
+      jobRunID,
+      message: `Input, at 'symbol' path, must be one of the following values: ${keys}`,
+      statusCode: 400,
+    })
+  }
+
+  const result = {
+    coin: symbol,
+    addresses: addresses.map((address: Address) => ({ address })),
+  }
```

Comment:
It doesn't look like this matches what the README shows.

To address part of @justinkaseman's request, I think something like this would be better suited:

```suggestion
  const result = addresses.map((address: Address) => ({ address, coin: symbol }))
```

Reply:
I think that since it is filtered by symbol, there is no need to specify the symbol in each address. Putting it as a separate attribute in the answer works best.

```  
"result": {
        "coin": "ETH",
        "addresses": [
            {
                "address": "0x6E4C739B4d66E7104739Fc0F235E9c9FFD1F8da4"
            },
            {
                "address": "0x67a53ADbA557E6129D3Eb1aab090E23D22711DB2"
            },
            {
                "address": "0x57Fc3f242aD907E9518645Ba4D0802F7a65B3677"
            }
        ]
    },
```
instead of returning: 
```  
"result": {
        "addresses": [
            {
                "address": "0x6E4C739B4d66E7104739Fc0F235E9c9FFD1F8da4",
                "coin: "ETH"
            },
            {
                "address": "0x67a53ADbA557E6129D3Eb1aab090E23D22711DB2",
                "coin: "ETH"
            },
            {
                "address": "0x57Fc3f242aD907E9518645Ba4D0802F7a65B3677",
                "coin: "ETH"
            }
        ]
    },
```

Reply:
I had a chat with Justin and Dimitri about this yesterday. We should make the adapter return an array of this interface:
```
{
  address: string // e.g. "0x123abc..."
  network: string // e.g. "ethereum"
  chainId: string | number // e.g "1" - or in cases of no chainId: "mainnet"
}
```

So in the case of your example above, we should expect to get this back:

```
"result": [
            {
                "address": "0x6E4C739B4d66E7104739Fc0F235E9c9FFD1F8da4",
                "network": "ethereum",
                "chainId": 1
            },
            {
                "address": "0x67a53ADbA557E6129D3Eb1aab090E23D22711DB2",
                "network": "ethereum",
                "chainId": 1
            },
            {
                "address": "0x57Fc3f242aD907E9518645Ba4D0802F7a65B3677",
                "network": "ethereum",
                "chainId": 1
            }
        ],
```

Since the Wrapped.com data source doesn't provide enough information itself to determine the network and chainId, the adapter should take `network` and `chainId` as required input params.